### PR TITLE
EVSE and Device Energy Management XML should use new base energy reporting types (and other minor fixes)

### LINKED ISSUE #30665: [1.3] Updates needed on EVSE XML & Device Energy Management - to use new types, and clean up other comments with optional XML.
### Reproduction steps

EVSE XML follow up issues based on PR: #30581 

- Please ensure a follow-up issue is filed to add the energy-mWh, amperage-mA, etc types to the XML definitions.
- For what it's worth, "max" is ignored here for readonly attributes, and the "is nullable and of type epoch_s" defines that max anyway. It doesn't hurt things, other than visual noise, but is not needed either. Spec does not have it (because it's part of the type already).
-This is the default; does not need to be explicitly listed, but ok to do it:
-     <access op="invoke" privilege="operate"/> - 
-  There is no "side" for events, right? All events are server-side....
-     <event code="0x0000" name="EVConnected" priority="info" side="server" apiMaturity="provisional">

### Bug prevalence

N/A

### GitHub hash of the SDK that was being used

Will need to be fixed in 1.3

### Platform

core

### Platform Version(s)

N/A

### Type

Spec Compliance Issue

### Anything else?

_No response_

### LINKED ISSUE #30805: [EVSE] SessionID null value XML change
Implement the changes to make Session attributes Nullable by updating XML.
Also change TargetTime -> TargetTimeMinutesPastMidnght

### LINKED ISSUE #31032: EVSE cluster now won't allow events to be turned on in ZAP
### Reproduction steps

1. Run ./scripts/tools/zap/run_zaptool.sh examples/energy-management-app/energy-management-common/energy-management-app.zap
2. Goto EP1, and Energy EVSE cluster
3. Click on Events tab
4. Turn on an event

Nothing happens but an error message is sent on console:

```(node:157908) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'endpointTypeClusterId')

    at Object.insertOrUpdateEventState (/home/jebh/workspace/connectedhomeip/.environment/cipd/packages/zap/resources/app.asar/dist/src-electron/db/query-config.js:384:29)

    at /home/jebh/workspace/connectedhomeip/.environment/cipd/packages/zap/resources/app.asar/dist/src-electron/rest/user-data.js:298:9

(Use `zap --trace-warnings ...` to show where the warning was created)

(node:157908) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)

```

I think this is because I was asked to remove side="server" from events in XML.
(ISSUE here: https://github.com/project-chip/connectedhomeip/issues/30665) @bzbarsky-apple 

```
-    <event code="0x0000" name="EVConnected" priority="info" side="server" apiMaturity="provisional">
+    <event code="0x0000" name="EVConnected" priority="info" apiMaturity="provisional">
```

### Bug prevalence

Every time

### GitHub hash of the SDK that was being used

N/A see PR #30857

### Platform

core

### Platform Version(s)

1.3

### Type

Spec Compliance Issue

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jamesharrow**
Fixes #30665
- https://github.com/project-chip/connectedhomeip/issues/30665

Fixes #30805 
- https://github.com/project-chip/connectedhomeip/issues/30805

energy-evse-server.xml:
- Changed to use amperage_ma, energy_mwh
- removed max on epoch_s
- removed access for operate
- removed side for events

Updated Device Energy Management:
- Used energy_mwh instead of int64_t
- Used power_mw instead of int64_t

for #30805 (EVSE Session attributes should be nullable)
- Made Session attributes Nullable
- Made SessionID in Fault event Nullable

Fixes #31032 
- Reverted change in events "side="server"' which was removed previously.

### CODE REVIEW

File: zzz_generated/chip-tool/zap-generated/cluster/Commands.h [hunk 1]
```diff
@@ -6874,7 +6874,7 @@ class DeviceEnergyManagementPowerAdjustRequest : public ClusterCommand
     DeviceEnergyManagementPowerAdjustRequest(CredentialIssuerCommands * credsIssuerConfig) :
         ClusterCommand("power-adjust-request", credsIssuerConfig)
     {
-        AddArgument("Power", INT64_MIN, INT64_MAX, &mRequest.power);
+        AddArgument("Power", 0, UINT64_MAX, &mRequest.power);
```
Comment (jamesharrow) [inline zzz_generated/chip-tool/zap-generated/cluster/Commands.h:6877]: Why has ZAP mad this unsigned? Is this a bug in ZAP?

This is the XML which defined Power as "power_mw":

```
    <command source="client" code="0x0000" name="PowerAdjustRequest" optional="true" apiMaturity="provisional">
      <arg name="Power" type="power_mw"/>
      <arg name="Duration" type="elapsed_s"/>
      <description>Allows a client to request an adjustment in the power consumption of an ESA for a specified duration.</description>
    </command>
```

File: zzz_generated/chip-tool/zap-generated/cluster/Commands.h [hunk 2]
```diff
@@ -7238,8 +7238,8 @@ class EnergyEvseEnableCharging : public ClusterCommand
     EnergyEvseEnableCharging(CredentialIssuerCommands * credsIssuerConfig) : ClusterCommand("enable-charging", credsIssuerConfig)
     {
         AddArgument("ChargingEnabledUntil", 0, UINT32_MAX, &mRequest.chargingEnabledUntil);
-        AddArgument("MinimumChargeCurrent", INT64_MIN, INT64_MAX, &mRequest.minimumChargeCurrent);
-        AddArgument("MaximumChargeCurrent", INT64_MIN, INT64_MAX, &mRequest.maximumChargeCurrent);
+        AddArgument("MinimumChargeCurrent", 0, UINT64_MAX, &mRequest.minimumChargeCurrent);
```
Comment (jamesharrow) [inline zzz_generated/chip-tool/zap-generated/cluster/Commands.h:7241]: Same here

File: zzz_generated/chip-tool/zap-generated/cluster/Commands.h (hunk 1 above)
Comment (bzbarsky-apple) [reply zzz_generated/chip-tool/zap-generated/cluster/Commands.h:6877]: Seems like a ZAP bug, yes.  The template has: `{{as_type_min_value type language='c++'}}, {{as_type_max_value type language='c++'}}` but it's clearly ending up with the wrong things...

Comment (bzbarsky-apple) [reply zzz_generated/chip-tool/zap-generated/cluster/Commands.h:6877]: @jamesharrow You want `signed="true"` on your type definitions, no?

Comment (jamesharrow) [reply zzz_generated/chip-tool/zap-generated/cluster/Commands.h:6877]: Ah - yes thanks that has fixed it.

**Review by brdandu — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/device-energy-management-cluster.xml [hunk 3]
```diff
@@ -38,14 +38,14 @@ limitations under the License.
     <attribute code="0x0000" side="server" type="ESATypeEnum" define="ESA_TYPE" writable="false" optional="false">ESAType</attribute>
     <attribute code="0x0001" side="server" type="boolean" define="ESA_CAN_GENERATE" writable="false" optional="false">ESACanGenerate</attribute>
     <attribute code="0x0002" side="server" type="ESAStateEnum" define="ESA_STATE" default="0" writable="false" optional="false">ESAState</attribute>
-    <attribute code="0x0003" side="server" type="int64s" define="ABS_MIN_POWER" default="0" writable="false" optional="false">AbsMinPower</attribute>
-    <attribute code="0x0004" side="server" type="int64s" define="ABS_MAX_POWER" default="0" writable="false" optional="false">AbsMaxPower</attribute>
+    <attribute code="0x0003" side="server" type="power_mw" define="ABS_MIN_POWER" default="0" writable="false" optional="false">AbsMinPower</attribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/device-energy-management-cluster.xml:41]: I thought we were calling these `power_mW` and so on?

Anyway, this looks fine.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/device-energy-management-cluster.xml (hunk 3 above)
Comment (jamesharrow) [reply src/app/zap-templates/zcl/data-model/chip/device-energy-management-cluster.xml:41]: I suggest we do that as a follow on - there was a weird changes in Java autogen if you recall.

**Pull request merged**